src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.order.CreateOrderRequestDto;
+import com.example.bookstore.dto.order.OrderDto;
+import com.example.bookstore.dto.order.OrderWithoutItemsDto;
+import com.example.bookstore.dto.order.UpdateOrderDto;
+import com.example.bookstore.dto.order.items.OrderItemDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.OrderMapper;
+import com.example.bookstore.model.Order;
+import com.example.bookstore.model.OrderItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.OrderItemRepository;
+import com.example.bookstore.repository.OrderRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.service.OrderItemService;
+import com.example.bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemService orderItemService;
+    private final UserDetailsService userDetailsService;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public OrderWithoutItemsDto createOrder(Authentication authentication,
+                                            CreateOrderRequestDto orderRequestDto) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
```

Comment:
as alternative you can use `User user = (User) authentication.getPrincipal();`

Reply:
It looks much better, but honestly, I don't want to change it everywhere. But if it is bad practice, I will